components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java

```diff
@@ -204,80 +216,86 @@ private DropboxFileUploadResult putBody(Exchange exchange, DropboxUploadMode mod
         return result;
     }
 
-    private DbxEntry.File putSingleFile(File inputFile, String dropboxPath, DropboxUploadMode mode) throws Exception {
+    private FileMetadata putSingleFile(File inputFile, String dropboxPath, DropboxUploadMode mode) throws Exception {
         FileInputStream inputStream = new FileInputStream(inputFile);
-        DbxEntry.File uploadedFile;
+        FileMetadata uploadedFile;
         try {
-            DbxWriteMode uploadMode;
+            WriteMode uploadMode;
             if (mode == DropboxUploadMode.force) {
-                uploadMode = DbxWriteMode.force();
+                uploadMode = WriteMode.OVERWRITE;
             } else {
-                uploadMode = DbxWriteMode.add();
+                uploadMode = WriteMode.ADD;
             }
-            uploadedFile = client.uploadFile(dropboxPath, uploadMode, inputFile.length(), inputStream);
+            uploadedFile = client.files().upload(dropboxPath).uploadAndFinish(inputStream, inputFile.length());
             return uploadedFile;
         } finally {
             IOHelper.close(inputStream);
         }
     }
 
-    private DbxEntry.File putSingleBody(Exchange exchange, String dropboxPath, DropboxUploadMode mode) throws Exception {
+    private FileMetadata putSingleBody(Exchange exchange, String dropboxPath, DropboxUploadMode mode) throws Exception {
         byte[] data = exchange.getIn().getMandatoryBody(byte[].class);
         InputStream is = new ByteArrayInputStream(data);
         try {
-            DbxEntry.File uploadedFile;
-            DbxWriteMode uploadMode;
+            FileMetadata uploadedFile;
+            WriteMode uploadMode;
             if (mode == DropboxUploadMode.force) {
-                uploadMode = DbxWriteMode.force();
+                uploadMode = WriteMode.OVERWRITE;
             } else {
-                uploadMode = DbxWriteMode.add();
+                uploadMode = WriteMode.ADD;
             }
-            uploadedFile = client.uploadFile(dropboxPath, uploadMode, data.length, is);
+            uploadedFile = client.files().upload(dropboxPath).uploadAndFinish(is, data.length);
```

Comment:
Use uploadBuilder which allows to use the uploadMode

Reply:
Ack, done